Realm/RLMNetworkTransport.mm

```diff
@@ -198,6 +199,13 @@ + (instancetype)delegateWithEventSubscriber:(RLMEventSubscriber *)subscriber {
 - (void)URLSession:(__unused NSURLSession *)session
           dataTask:(__unused NSURLSessionDataTask *)dataTask
     didReceiveData:(NSData *)data {
+    {
+        std::lock_guard<std::mutex> lock(_eventMutex);
```

Comment:
NSURLSessionDelegate methods are invoked on a serial queue so there shouldn't be any need for a mutex here.

Reply:
Ah, good to know, will remove 👍